src/components/filter/filter.tsx

```diff
@@ -179,10 +179,14 @@ export class Filter implements InteractiveComponent {
     this.filter(target.value, true);
   };
 
-  keyDownHandler = ({ key }: KeyboardEvent): void => {
-    if (key === "Escape") {
+  keyDownHandler = (event: KeyboardEvent): void => {
+    if (event.key === "Escape") {
       this.clear();
```

Comment:
todo: this should be preventable in separate issue/PR

Reply:
Is this suggestion to be able to prevent the filter from being cleared when the event is canceled?

Reply:
Yes, correct.

Reply:
We could add the event listener on the document using the `@Event` decorator and check to see if it has `event.defaultPrevented` in this case as a solution to keep the logic on this component.

Reply:
https://github.com/Esri/calcite-components/issues/4507